src/map/script.cpp

```diff
@@ -5758,18 +5764,34 @@ BUILDIN_FUNC(warpparty)
 			x = pl_sd->bl.x;
 			y = pl_sd->bl.y;
 			break;
-		case 4:
+		case 4: {
+			if ( !script_rid2sd(sd) )
+				return SCRIPT_CMD_SUCCESS;
+
+			mapindex = sd->mapindex;
+			m = map_mapindex2mapid(mapindex);
+
+			struct map_data *mapdata = map_getmapdata(m);
+
+			if ( mapdata->flag[MF_NOWARP] || mapdata->flag[MF_NOTELEPORT] )
+				return SCRIPT_CMD_FAILURE;
```

Comment:
Maybe add a warning and a check of the mapflags on the script if you return SCRIPT_CMD_FAILURE

Reply:
That would show a warning each time someone uses a giant fly wing in a map with those mapflags

Reply:
Yes but you may add a check of the mapflag in the script.
Eventually replace SCRIPT_CMD_FAILURE by SCRIPT_CMD_SUCCESS, if I'm not wrong SCRIPT_CMD_FAILURE throw a debug message containing the npc path

Reply:
Ah, you're right, but I'll leave the mapflag check here, because this command has similar checks for the other types a few lines below. I'll just replace the return value.